cordova-lib/src/cordova/restore-util.js

```diff
@@ -64,7 +64,25 @@ function installPlatformsFromConfigXML(platforms, opts) {
             // Combining arrays and checking duplicates.
             comboArray = pkgJsonPlatforms.slice();
         }
+
         engines = cfg.getEngines(projectHome);
+
+        // TODO: CB-12592: Eventually refactor out to pacakge manager module.
+        // If package.json doesn't exist, auto-create one.
+        if (engines.length > 0 && !fs.existsSync(pkgJsonPath)) {
```

Comment:
Was able to remove the duplicated lines:
```    if(fs.existsSync(pkgJsonPath)) {
        pkgJson = require(pkgJsonPath);```

But if you update restore-util from:
```} else if (engines.length > 0 && !fs.existsSync(pkgJsonPath)) {```
to
```} else if (engines.length > 0 && pkgJson) {```
there will be many failures... will look into them